plugins/inputs/nfsclient/nfsclient.go

```diff
@@ -0,0 +1,421 @@
+		}
+		acc.AddFields("nfs_events", fields, tags)
+	} else if fullstat && first == "bytes" && len(nline) >= len(bytesFields) {
+		for i, t := range bytesFields {
+			fields[t] = nline[i]
+		}
+		acc.AddFields("nfs_bytes", fields, tags)
+	} else if fullstat && first == "xprt" && len(line) > 1 {
+		switch line[1] {
+		case "tcp":
+			if len(nline)+2 >= len(xprttcpFields) {
+				for i, t := range xprttcpFields {
+					fields[t] = nline[i+2]
+				}
+				acc.AddFields("nfs_xprt_tcp", fields, tags)
+			}
+		case "udp":
+			if len(nline)+2 >= len(xprtudpFields) {
+				for i, t := range xprtudpFields {
+					fields[t] = nline[i+2]
+				}
+				acc.AddFields("nfs_xprt_udp", fields, tags)
+			}
+		}
+	} else if fullstat {
+		if version == "3" {
+			if nfs3Ops[first] && (len(nline) <= len(nfsopFields)) {
+				for i, t := range nline {
+					item := fmt.Sprintf("%s_%s", first, nfsopFields[i])
+					fields[item] = t
+				}
+				acc.AddFields("nfs_ops", fields, tags)
+			}
+		} else if version == "4" {
+			if nfs4Ops[first] && (len(nline) <= len(nfsopFields)) {
+				for i, t := range nline {
+					item := fmt.Sprintf("%s_%s", first, nfsopFields[i])
+					fields[item] = t
+				}
+				acc.AddFields("nfs_ops", fields, tags)
+			}
+		}
+	} else {
+		if first == "READ" {
+			fields["read_ops"] = nline[0]
+			fields["read_retrans"] = (nline[1] - nline[0])
+			fields["read_bytes"] = (nline[3] + nline[4])
+			fields["read_rtt"] = nline[6]
+			fields["read_exe"] = nline[7]
+			acc.AddFields("nfsstat_read", fields, tags)
+		} else if first == "WRITE" {
+			fields["write_ops"] = nline[0]
+			fields["write_retrans"] = (nline[1] - nline[0])
+			fields["write_bytes"] = (nline[3] + nline[4])
+			fields["write_rtt"] = nline[6]
+			fields["write_exe"] = nline[7]
+			acc.AddFields("nfsstat_write", fields, tags)
+		}
+	}
+
+	return nil
+}
+
+func (n *NFSClient) processText(scanner *bufio.Scanner, acc telegraf.Accumulator) error {
+	var device string
+	var version string
+	var export string
+	var skip bool
+	var nfs3Ops map[string]bool
+	var nfs4Ops map[string]bool
+	nfs3Ops = make(map[string]bool)
+	nfs4Ops = make(map[string]bool)
+
+	if len(n.IncludeOperations) == 0 {
+		for _, Op := range nfs3Fields {
+			nfs3Ops[Op] = true
+		}
+		for _, Op := range nfs4Fields {
+			nfs4Ops[Op] = true
+		}
+	} else {
+		for _, Op := range n.IncludeOperations {
+			nfs3Ops[Op] = true
+		}
+		for _, Op := range n.IncludeOperations {
+			nfs4Ops[Op] = true
+		}
+	}
+
+	if len(n.ExcludeOperations) > 0 {
+		for _, Op := range n.ExcludeOperations {
+			if nfs3Ops[Op] {
+				delete(nfs3Ops, Op)
+			}
+			if nfs4Ops[Op] {
+				delete(nfs4Ops, Op)
+			}
+		}
+	}
+
+	for scanner.Scan() {
+		line := strings.Fields(scanner.Text())
+
+		if in(line, "fstype") && (in(line, "nfs") || in(line, "nfs4")) && len(line) > 4 {
+			device = line[4]
+			export = line[1]
+			skip = false
+		} else if (in(line, "(nfs)") || in(line, "(nfs4)")) && len(line) > 5 {
+			version = strings.Split(line[5], "/")[1]
+		}
+
```

Comment:
Can we please initialize `skip` here?!

Reply:
See 8a78470359de3c050262bf50208484c0783825b9

Reply:
The linked commit does not exist anymore. What are your assumptions in the following code? If I understand your intention correctly, you assume `skip ==  false`  in your filtering logic, which is correct for the initial `for loop` execution. That is, if there are **no** `IncludeMounts` specified, you evaluate the excludes. But what happens in case you have multiple mounts and the first one is in `ExcludeMounts`? The `skip` will be set to `true` and stays `true` for the following mounts! Remember `IncludeMounts` is empty (which is not very constructed either) and thus `skip` is `true` for all following mounts even tough those mounts are **not** in the exclude list!!! This way you will get no data at all. :-(

Reply:
Can you please comment on this one?